nltk/translate/lepor.py

```diff
@@ -0,0 +1,253 @@
+                    and ref_words[ref_word_index - 1] == hypothesis_words[hyp_index - 1]
+                ):
+                    is_matched[ind] = True
+                elif (
+                    0 < ref_word_index + 1 < len(ref_words)
+                    and 0 < hyp_index + 1 < len(hypothesis_words)
+                    and ref_words[ref_word_index + 1] == hypothesis_words[hyp_index + 1]
+                ):
+                    is_matched[ind] = True
+
+            if is_matched.count(True) == 1:
+                alignments.append(ref_indexes[is_matched.index(True)])
+            elif is_matched.count(True) > 1:
+                min_distance = 0
+                min_index = 0
+                for match, ref_index in zip(is_matched, ref_indexes):
+                    if match:
+                        distance = abs(hyp_index - ref_index)
+                        if distance > min_distance:
+                            min_distance = distance
+                            min_index = ref_index
+                alignments.append(min_index)
+
+            else:
+                min_distance = 0
+                min_index = 0
+                for ref_index in ref_indexes:
+                    distance = abs(hyp_index - ref_index)
+                    if distance > min_distance:
+                        min_distance = distance
+                        min_index = ref_index
+                alignments.append(min_index)
+
+    alignments = [a + 1 for a in alignments if a != -1]
+    match_count = len(alignments)
+    npd_list = []
+
+    for ind, a in enumerate(alignments):
+        npd_list.append(abs(((ind + 1) / len(hypothesis_words)) - (a / len(ref_words))))
+    npd = sum(npd_list) / len(hypothesis_words)
+
+    return math.exp(-npd), match_count
+
+
+def harmonic(
+    match_count: int,
+    reference_length: int,
+    hypothesis_length: int,
+    alpha: float,
+    beta: float,
+) -> float:
+    """
+    Function will calculate the precision and recall of matched words and calculate a final score on wighting
+    using alpha and beta parameters.
+
+    :param match_count: number of words in hypothesis aligned with reference.
+    :type match_count: int
+    :param reference_length: length of the reference sentence
+    :type reference_length: int
+    :param hypothesis_length: length of the hypothesis sentence
+    :type hypothesis_length: int
+    :param alpha: a parameter to set weight fot recall.
+    :type alpha: float
+    :param beta: a parameter to set weight fot precision.
+    :type beta: float
+
+    :return: harmonic mean.
+    :rtype: float
+    """
+
+    epsilon = sys.float_info.epsilon
+
+    precision = match_count / hypothesis_length
+    recall = match_count / reference_length
+
+    harmonic_score = (alpha + beta) / (
+        (alpha / (recall + epsilon)) + (beta / (precision + epsilon))
+    )
+
+    return harmonic_score
+
+
+def sentence_lepor(
+    reference: str, hypothesis: str, alpha: float = 1.0, beta: float = 1.0
```

Comment:
The `sentence_lepor` should follow the general interface of the other metrics to take in a list of references per hypothesis.

Reply:
Yes, this is the case with other metrics. i will provide the same facility as well.